pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -427,6 +429,14 @@ cdef inline parse_timedelta_string(object ts):
             have_value = 1
             have_dot = 0
 
+    # Consider units from outside
+    if not unit:
+        if specified_unit:
+            unit = [specified_unit]
+    else:
+        if specified_unit:
+            raise ValueError("units doubly specified")
```

Comment:
can you make this a better error message (and include the parsed unit and the passed unit)